dataservice/api/common/schemas.py

```diff
@@ -19,18 +24,47 @@ class Meta:
         exclude = ('uuid',)
         dump_only = ('created_at', 'modified_at')
 
+    @pre_dump(pass_many=True)
+    def wrap_pre(self, data, many):
+        if isinstance(data, Pagination):
+            self.__pagination__ = data
+            return data.items
+        return data
+
     @post_dump(pass_many=True)
     def wrap_envelope(self, data, many):
+        """
+        Wraps ORM objects in a standard response format
+
+        If `many=False`, only one object is being returned and any `_links`
+        that are in the data are moved out to the response's `_links`.
+
+        If `many=True`, many objects are being returned and the top `_links`
+        in the response will be populated with pagination details.
+        """
+        resp = {'_status': {'message': self.status_message,
+                            'code': self.status_code}}
+        # Move links to the envelope links if just a single object
         if not many and '_links' in data:
             _links = data['_links']
             del data['_links']
+        # Insert pagination object, if there is one
+        elif many and self.__pagination__ is not None:
```

Comment:
Should a link to prev be in here too?

Reply:
Only supports iteration in one direction, at the moment.